modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/delegate/ParameterModelsLoaderDelegate.java

```diff
@@ -64,9 +76,32 @@ public List<ParameterDeclarer> declare(HasParametersDeclarer component, List<Par
               .defaultingTo(parameterParser.getDefaultValue());
         }
 
+        Optional<KeyIdResolverModelParser> parameterKeyIdResolverModelParser = parameterParser.getKeyIdResolverModelParser(null);
+        Optional<Pair<Integer, Boolean>> metadataKeyPart = parameterParser.getMetadataKeyPart();
+
+        if (parameterKeyIdResolverModelParser.isPresent() && !parameterKeyIdResolverModelParser.get().hasKeyIdResolver()
+            && parentKeyIdResolverModelParser != null) {
+          parameterKeyIdResolverModelParser = of(new JavaKeyIdResolverModelParser(
```

Comment:
Same as before, The loader delegate must not create jave specific implementations of the parsers.

Reply:
Also, it seems to be a smell that the declare method had to be overloaded to get the new KeyIdResolverModelParser